### DISCUSSION TITLE
Wrong metric type ?

### DISCUSSION THREAD

**Discussion by phibou60** [open-telemetry/opentelemetry-java-instrumentation#11279] [Q&A] [answered]
Hello, I use the java agent 2.3.0.
Maybe I’m wrong but I think that some metrics should be reported in “gauge” type rather than “sum”.

For example, the metric "jvm.memory.used" is the memory usage at a point of time.
This follow the definition of a gauge : Gauge represents the type of a scalar metric that always exports the "current value" for every data point. (comment from metrics.proto)

But the metric is reported as a "sum".
The problem is that our visualization tool aggregates the values sometimes.

For example, if we choose to visualize a big interval like one month of data, our tool simplifies the data by aggregating by hour.
For gauge metrics, there should be no addition, and a single value is chosen for the interval of an hour.
But for sum metrics, the tool add all the values in the interval of a hour giving wrong values for the memory used.

**Comment by laurit** [comment]
See https://opentelemetry.io/docs/specs/otel/metrics/supplementary-guidelines/#instrument-selection `jvm.memory.used` is [additive](https://opentelemetry.io/docs/specs/otel/metrics/supplementary-guidelines/#additive-property).

**Reply by phibou60** [reply, marked as answer]
Thanks you for the link.
I've found a workaround.
